libraries/AP_AccelCal/AccelCalibrator.cpp

```diff
@@ -333,6 +333,7 @@ void AccelCalibrator::run_fit(uint8_t max_iterations, float& fitness)
     union param_u fit_param = _param;
     uint8_t num_iterations = 0;
 
+    MatInvBuffer<float> buffer {ACCEL_CAL_MAX_NUM_PARAMS*ACCEL_CAL_MAX_NUM_PARAMS};
```

Comment:
So if allocation fails here then we just rumble on?

I haven't spotted handling within the new class for that - but not handling the allocation failure here seems bad.

Reply:
Hi, that is a good question that I wanted to bring to the review. The original mat_inverse implementation has the same issue,  there are 7 allocations with unhandled failures: ([link](https://github.com/ArduPilot/ardupilot/blob/60c2e4278cc38591187b8a188677bd66ac8f210a/libraries/AP_Math/matrix_alg.cpp#L202)).

Fixing this would require more changes in the dependent modules, so I guess it could be done in a separate pr. I can take that task too, but I'm new to ArduPilot, and I need some support in clarifying the recovery strategy for allocation failures.

Reply:
I guess the solution is to reserve memory on client class initialization and handle failures accordingly on that step. Another option is to place it on stack, with the price of increasing frame size.